server/controllers/providerReportController.ts

```diff
@@ -0,0 +1,108 @@
+import type { Request, RequestHandler, Response } from 'express'
```

Comment:
There is a lot of duplicated code in here from generateReportController. I think we should only have 1 controller  so we have this:- /provider-report -> generateReportController. Then generateReportController should be able to determine whether this is for a provider report and act accordingly.